doc/development/style_guide.md

```diff
@@ -0,0 +1,203 @@
+
+- Use markdown documentation in `doc/` folder to give the high-level overview of the functionality,
+  and especially how it fits into the overall Flame framework.
+
+
+### Dartdocs
+
+- Check the [Flutter Documentation Guide] -- it contains lots of great advice on writing good
+  documentation.
+    - However, disregard the advice about writing in a passive voice.
+
+- Class documentation should ideally start with the class name itself, and follow a pattern such as:
+  ```dart
+  /// [MyClass] is ...
+  /// [MyClass] serves as ...
+  /// [MyClass] does the following ...
+  ```
+  The reason for such convention is that often the documentation for a class becomes sufficiently
+  long, and it may not be immediately apparent when looking at the top of the doc what exactly is
+  being documented there.
+
+- Method documentation should start with a verb in the present simple tense, with the method name
+  as an implicit subject. Add a paragraph break after the first sentence. Try to think about what
+  could be unclear to the users of the method; and mention any pre- and post-conditions.
+  For example:
+  ```dart
+  /// Adds a new [child] into the container, and becomes the owner of that child.
+  ///
+  /// The child will be disposed of when this container is destroyed. It is an error to try to add a
+  /// child that already belongs to another container.
+  void addChild(T child) { ... }
+  ```
+  Avoid stating the obvious (or at least only the obvious).
+
+- Constructor documentation may follow either the style of a method (i.e. "Creates ...",
+  "Constructs ..."), or of the class but omitting the name of the class (i.e. "Rectangular-shaped
+  component"). Constructor documentation may be omitted if (1) it's the main constructor of the
+  class, and (2) all the parameters are obvious and coincide with the public members of the class.
+
+  **Do not** use macros to copy the class documentation into the constructor's dartdoc. Generally,
+  the class documentation answers the question "what the class is", whereas the constructor docs
+  answer "how it will be constructed".
+
+
+### Main docs
+
+This refers to the docs on the main Flame Documentation website, the one you're reading right now.
+The main documentation site serves as a place where people go to learn about various functionality
+available in Flame. If you're adding a new class, then it must be documented both at the dartdocs
+level, and on the main docs site. The latter serves the purposes of discoverability of your class.
+Without the docs site, your class might never be used (or at least used less than it could have
+been), because the developers would simply not know about it.
+
+When adding the documentation to the main docs site, consider also including an example directly
+into the docs. This will make readers more excited about trying this new functionality.
+
+Check the [Documentation] manual about how to work with the docs site.
```

Comment:
Should we add a few sentences about the markdown structures that we are following too? Like two lines before a new section, max 100 chars on a row etc?

Reply:
Right, I forgot about those. Will add.

Reply:
Done.